libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/ptx_isa_target_macros.h

```diff
@@ -0,0 +1,63 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef PTX_ISA_TARGET_MACROS_H_
+#define PTX_ISA_TARGET_MACROS_H_
+
+
+/*
+ * Targeting macros
+ *
+ * Information from:
+ * https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#release-notes
+ */
+
+#if (defined(__CUDA_MINIMUM_ARCH__) && 800 <= __CUDA_MINIMUM_ARCH__) || (!defined(__CUDA_MINIMUM_ARCH__))
+#  define _LIBCUDACXX_PTX_SM_80_AVAILABLE
+#endif
+
+#if (defined(__CUDA_MINIMUM_ARCH__) && 900 <= __CUDA_MINIMUM_ARCH__) || (!defined(__CUDA_MINIMUM_ARCH__))
+#  define _LIBCUDACXX_PTX_SM_90_AVAILABLE
```

Comment:
There is a considerable chance that we want to reuse them and that we also want to have our users rely on them.

Therefore, I would suggest to name them as `CCCL_PTX_ISA_78_AVAILABLE`

This would be in line with us going in the direction a unified of `CCCL` repo rather than just libcu++ and also makes it clear that those are user facing defines

Reply:
By your suggestion, I changed the names to `__cccl_ptx_sm, __cccl_ptx_isa`. In addition, macro values are now numeric. 

> I would love to give users something they can reliably test.

Re: reliability. These macros are only accurate in device code (when compiled by nvcc). In host code, they are unreliable (by design).